src/RunAction.cpp

```diff
@@ -71,28 +111,34 @@ void RunAction::BeginOfRunAction(const G4Run* run)
         // G4cout << "stem:           " << path.stem()           << G4endl;
         // G4cout << "extension:      " << path.extension()      << G4endl;
 
-        std::string parent_path = path.parent_path();
-        std::string stem = path.stem();
-        std::string extension = path.extension();
+        G4String parentPath_ = outputFile_(0, outputFile_.last('/'));
```

Comment:
I am having issues to compile the code with my Geant4 version (11.1.0) as the member `last` in `G4String` seems to have been removed from version 4.11 onwards.

Reply:
@asanchezcastillo Is qpixg4 supposed to be working with Geant4 11 at all?  My own experience (with the SuperCDMS simulation), is that if you migrate code to compile and run with G4 11, then it will not compile with G4 10.7, and vice versa.

I thought Q-Pix had decided (at least for a while) to stick with G4 10.7, to avoid those problems.

Reply:
@kelseymh there is a google doc containing an installation guide for all the qpix software in which G411.1.0 is used. I also found out in the qpixg4 wiki one of the dependencies for qpixg4-v2.0.0 is Geant4 11.0+. I particularly do not mind using G410.7 but I do not know if it could be inconvenient for other people.

Reply:
Ah, thank you!  Then I was mistaken.  I do see that `.last()` was removed from G4String with G4 11.  @delofson0211 You may want to look at replacing `.last()` with a use of `.rbegin()` or something directly from `std::string` (which is the base class of `G4String`).

Reply:
@asanchezcastillo qpixg4 is made specifically to be compatible with Geant4.10.7, and not with Geant4.11. There are many breaking changes. The wiki needs to be changed. Making it compatible with Geant4.11 will be a project for later (and someone else)

Reply:
@delofson0211 If this branch of qpixg4 does build and run against G4 11, this sounds like a small fix to keep it that way.  If it doesn't, then what you say above is exactly right.

Reply:
Yes. There are no branches that do run against Geant4.11. It was supposed to be an easy fix, but it turned out to be much more complicated.